### PR TITLE
Add ADNS9800 optical motion sensor driver.

### PR DESCRIPTION
**Pull request by strongly-typed**
Worked on my bench.

### CODE REVIEW

**Comment by salkinium** [comment]
> Worked on my bench.

is the new Works-for-me™. 😝

File: src/xpcc/driver/motion/adns9800_srom_a4.hpp [hunk 1]
```diff
@@ -0,0 +1,201 @@
+#ifndef ADNS_9800_SROM_A4_HPP
+#define ADNS_9800_SROM_A4_HPP
+
+static constexpr uint16_t firmware_length = 3070;
+
+static constexpr uint8_t firmware_data[firmware_length] =
```
Comment (ekiwi) [inline src/xpcc/driver/motion/adns9800_srom_a4.hpp:6]: could you add a comment that includes the version/source of this firmware blob?

Comment (salkinium) [reply src/xpcc/driver/motion/adns9800_srom_a4.hpp:6]: Is there a license associated with this firmware blob?

**Comment by strongly-typed** [comment]
PixArt/Avago Imaging provided the data  for download. Now the device is EOL and no new download is available. The blob is from project referenced in [adns9800.hpp](https://github.com/strongly-typed/xpcc/blob/feature/adns_9800/src/xpcc/driver/motion/adns9800.hpp#L12)

**Comment by salkinium** [comment]
Hm, the firmware was apparently once [hosted on the Avago website](https://github.com/mrjohnk/ADNS-9500/blob/master/firmwareA0.ino#L5-L7).

**Comment by salkinium** [comment]
It's probably fine™.

**Pull request merged**